scripts/scil_compute_connectivity.py

```diff
@@ -0,0 +1,351 @@
+
+    og = p.add_argument_group('Outliers and loops options')
+    og.add_argument('--outlier_threshold', type=float, default=0.3,
+                    help='Outlier removal threshold when using hierarchical '
+                         'QB. [%(default)s]')
+    og.add_argument('--loop_max_angle', type=float, default=360.,
+                    help='Maximal winding angle over which a streamline is '
+                         'considered as looping. [%(default)s]')
+    og.add_argument('--loop_qb_distance', type=float, default=15.,
+                    help='Maximal distance to a centroid for loop / turn '
+                         'filtering with QB. [%(default)s]')
+
+    s = p.add_argument_group('Saving options')
+    s.add_argument('--save_raw_connections', action='store_true',
+                   help='if set, will save all raw cut connections in a '
+                        'subdirectory')
+    s.add_argument('--save_intermediate', action='store_true',
+                   help='if set, will save the intermediate results of '
+                        'filtering')
+    s.add_argument('--save_discarded', action='store_true',
+                   help='if set, will save discarded streamlines in '
+                        'subdirectories.\nIncludes loops, outliers and '
+                        'qb_loops')
+
+    add_overwrite_arg(p)
+
+    p.add_argument('--verbose', '-v', dest='verbose',
+                   action='store_true', default=False,
+                   help='Verbose. [%(default)s]')
+    return p
+
+
+def main():
+    parser = build_args_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.tracts, args.labels])
+    assert_output_dirs_exist_and_empty(parser, args, args.output)
+
+    log_level = logging.WARNING
+    if args.verbose:
+        log_level = logging.INFO
+    logging.basicConfig(level=log_level)
+
+    img_labels = nb.load(args.labels)
+    if not np.issubdtype(img_labels.get_data_dtype().type, np.integer):
+        parser.error("Label image should contain integers for labels.")
+
+    # Ensure that voxel size is isotropic. Currently, for speed considerations,
+    # we take the length in voxel space and multiply by the voxel size. For
+    # this to work correctly, voxel size must be isotropic.
+    vox_sizes = img_labels.header.get_zooms()
+    if not np.mean(vox_sizes) == vox_sizes[0]:
+        parser.error('Labels must be isotropic')
+
+    if np.min(img_labels.get_data()) < 0 or \
+            np.max(img_labels.get_data()) > args.max_labels:
+        parser.error('Invalid labels in labels image')
+
+    streamlines = load_trk_in_voxel_space(args.tracts, args.labels)
+    logging.info('Number of streamlines to process: {}'.format(
+        len(streamlines)))
+
+    # Get all streamlines intersection indices
+    indices, points_to_idx = uncompress(streamlines, return_mapping=True)
+
+    # Compute the connectivity mapping
+    # TODO self connection?
```

Comment:
Do we remove this comment ?

Reply:
Well it should be implemented someday. Do you prefer creating an issue?